src/main/java/org/tools4j/hoverraft/command/DirectLogEntry.java

```diff
@@ -60,8 +67,17 @@ public CommandMessage commandMessage() {
     }
 
     @Override
+    public LogKey logKey() {
+        return directLogKey;
+    }
+
+    @Override
     public int byteLength() {
         return COMMAND_MSG_OFF + directCommandMessage.byteLength();
     }
 
+    @Override
+    public void copyFrom(final LogEntry logEntry) {
```

Comment:
that would only be possible for return values. not possible to subclass arguments.